### ISSUE TITLE
Transcoding using put to upload a file?

### ISSUE THREAD

**Issue by medz** [flutterchina/dio#619]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

How to use put to send a file to the server? My server only supports put and the body part is the file content. As Postman did:
![image](https://user-images.githubusercontent.com/5564821/71307697-f6714200-242c-11ea-87c3-599f147c020b.png)

**Comment by medz** [comment]
I gave up Dio. HttpClient is simpler. Dio can also PUT a stream directly. After carefully reading the source code, I find that the Data part only needs `File.openRand ()`. Really poor documentation! Goodbye Dio.
